ducktape/cluster/cluster.py

```diff
@@ -62,3 +66,20 @@ def __eq__(self, other):
 
     def __hash__(self):
         return hash(tuple(sorted(self.__dict__.items())))
+
+    @staticmethod
+    def _node_count_helper(nodes, operating_system):
+        count = 0
+        for node in nodes:
+            if node.operating_system == operating_system:
+                count += 1
+        return count
+
+    @staticmethod
+    def _next_available_node(nodes, operating_system):
+        node_to_return = None
+        for node in nodes:
+            if node.operating_system == operating_system:
+                node_to_return = node
```

Comment:
You could just return this immediately instead. Currently you'll iterate over the entire collection even if the first node you check matches. (You can also write these in a simpler more pythonic way, e.g. `return next(node for node in nodes if node.operating_system == operating_system)` for `_next_available_node` and `_node_count_helper` could be `return len([node for node in nodes if node.operating_system == operating_system])`.)